pkg/client/unversioned/request.go

```diff
@@ -648,12 +687,70 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+type URLBackoff struct {
```

Comment:
Very little of this code should be in this file - it should be separated out into its own file.